test/codeu/chat/server/DerbyStoreTest.java

```diff
@@ -0,0 +1,190 @@
+package codeu.chat.server;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import org.junit.Before;
+
+import codeu.chat.DerbyStore;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+import codeu.chat.common.BasicController;
+import codeu.chat.common.Conversation;
+import codeu.chat.common.Message;
+
+public class DerbyStoreTest {
+	
+	private DerbyStore ds;
+	private Model model;
+	private BasicController controller;
+		
+	@Before
+	public void doBefore() {
+		ds = new DerbyStore();
+		model = new Model();
+	    controller = new Controller(Uuid.NULL, model);
+	}
+	
+	@Test
+	public void checkUserNameExists() {
+		final User user = controller.newUser("user");
+		boolean userNameExists = false;
+		
+		try {
+			userNameExists = ds.checkUsernameExists(user.name);
+		}
+		catch (Exception ex) {
+			
+		}
+		
+		assertTrue("Check to see if a username already exists",
+				userNameExists);
+	}
+	
+	@Test
+	public void addAndGetUser() {
+		 final User user = controller.newUser("user");
+		 User testUser = null;
+		 try {
+			//ds.addUser(user);
+			 testUser = ds.getUser(user.id);
```

Comment:
Hmm okay, whatever you think is best is fine with me. I would just maybe expect to see two different levels: the database level that deals with low-level data access, and the controller level that sits on top of that and contains the logic. Right now they're a little mixed together.

Not saying you have to fix that right now, most of these comments are just stuff for you to think about. There isn't any single correct way to do most of this stuff!